numba/tests/test_stencils.py

```diff
@@ -813,127 +813,300 @@ def visit_Subscript(self, node):
+                if isinstance(node.ctx, ast.Store):
+                    msg = ("Assignments to array passed to "
+                            "stencil kernels is not allowed")
+                    raise ValueError(msg)
+                context = ast.Load()
+                newnode = ast.Subscript(
+                    value=node.value,
+                    slice=ast.Index(
+                        value=idx,
+                        ctx=ast.Load()),
+                    ctx=context)
+                ast.copy_location(newnode, node)
+                ast.fix_missing_locations(newnode)
+
+                # now work out max/min for index ranges i.e. stencil size
+                if self._mins is None and self._maxes is None:
+                    # first pass
+                    self._mins = [self._imax, ]
+                    self._maxes = [self._imin, ]
+
+                if not self._neighborhood:
+                    if isinstance(node.slice.value, ast.Num) or\
+                            isinstance(node.slice.value, ast.UnaryOp):
+                        relvalue = self.get_val_from_num(node.slice.value)
+                    elif (hasattr(node.slice.value, 'id') and
+                            node.slice.value.id in self._const_assigns):
+                        relvalue = self._const_assigns[node.slice.value.id]
                     else:
-                        if(self._idx_len != len(idx)):
-                            raise ValueError(
-                                "Relative indexing mismatch detected")
-                    if isinstance(node.ctx, ast.Store):
-                        msg = ("Assignments to array passed to "
-                               "stencil kernels is not allowed")
-                        raise ValueError(msg)
-                    context = ast.Load()
-                    newnode = ast.Subscript(
-                        value=node.value,
-                        slice=ast.Index(
-                            value=ast.Tuple(
-                                elts=idx,
+                        raise ValueError("Cannot interpret indexing value")
+                    if relvalue < self._mins[0]:
+                        self._mins[0] = relvalue
+                    if relvalue > self._maxes[0]:
+                        self._maxes[0] = relvalue
+                else:
+                    self._mins[0] = self._neighborhood[0][0]
+                    self._maxes[0] = self._neighborhood[0][1]
+                ast.copy_location(newnode, node)
+                ast.fix_missing_locations(newnode)
+                return newnode
+
+            def computeSlice(i, node):
+                def gen_idx(val, x):
+                    useval = self._const_assigns.get(val, val)
+                    value = self.get_val_from_num(val)
+                    tmp = ast.BinOp(
+                            left=ast.Name(
+                                id=self._id_pat %
+                                self.ids[x],
```

Comment:
This linebreak is odd

Reply:
https://github.com/numba/numba/pull/3719/commits/5dbca1ff3baac88354934043a83d74442b14b26d